File: ToLowerCase.java
/*Implement function ToLowerCase() that has a string parameter str, and returns the same string in lowercase.*/

class Solution {
   public String toLowerCase(String str) {
        if (str == null) {
            return null;
        }
        char chars[] = str.toCharArray();
        char target[] = new char[chars.length];
        int delta = 'a' - 'A';
        for (int i = 0; i < chars.length; i++) {
            if (chars[i] >= 'A' && chars[i] <= 'Z') {
                target[i] = (char) (chars[i] + delta);
            } else {
                target[i] = chars[i];
            }
        }
        return new String(target);
    }
}

File: ToLowerCase.py
/*Implement function ToLowerCase() that has a string parameter str, and returns the same string in lowercase.*/

class Solution:
    def toLowerCase(self, str):
        """
        :type str: str
        :rtype: str
        """
        strout=''
        for char in str:
            if ord(char)>64 and ord(char)<91:
                strout+=chr(ord(char)+32)
            else:
                strout+=char
        return strout
